src/Stack/Options.hs

```diff
@@ -861,7 +861,7 @@ ghcVariantParser hide =
         readGHCVariant
         (long "ghc-variant" <> metavar "VARIANT" <>
          help
-             "Specialized GHC variant, e.g. integersimple (implies --no-system-ghc)" <>
+             "Specialized GHC variant, e.g. integersimple" <>
```

Comment:
> That's still implied, no?

I guess that depends on how one understands "implies": I read it as "changes default settings" (in this case from `--system-ghc` to `--no-system-ghc`), while you seem to read it like "requires".

I haven't been able to test `--ghc-variant <sth-non-standard>` on Ubuntu-16.04 (I didn't try very hard) but according to #816 it's intended to be used only with `--no-system-ghc`.

I don't think `--ghc-variant` should silently override a user's `--system-ghc` setting, so how about I add  `(incompatible with --no-system-ghc)` to the CLI help line?
I'd also add a check that errors out if both `--ghc-variant X` and `--system-ghc` are passed in.